ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,462 @@
+                    (true, false) => Charset::Korea,
+                    (true, true) => Charset::Taiwan,
+                }
+            },
+        };
+
+        let position = SelectorPosition {
+            page_index: partial_u8_bits_to_u8(&raw_position[0..=3]),
+            slot_index: partial_u8_bits_to_u8(&raw_position[4..=7]),
+        };
+
+        let device = ConsoleIdentity {
+            origin_console: {
+                match (raw_device[6], raw_device[5], raw_device[4]) {
+                    (false, false, true) => OriginConsole::Wii,
+                    (false, true, false) => OriginConsole::DSi,
+                    (false, true, true) => OriginConsole::N3DS,
+                    _ => OriginConsole::WiiUSwitch,
+                }
+            },
+        };
+
+        let details = Details {
+            sex: {
+                match raw_details[0] {
+                    true => MiiSex::Female,
+                    false => MiiSex::Male,
+                }
+            },
+            birthday_month: partial_u8_bits_to_u8(&raw_details[1..=4]),
+            birthday_day: partial_u8_bits_to_u8(&raw_details[5..=9]),
+            shirt_color: partial_u8_bits_to_u8(&raw_details[10..=13]),
+            is_favorite: raw_details[14],
+        };
+
+        let face_details = FaceDetails {
+            style: FaceStyle {
+                is_sharing_enabled: !raw_face_style[0],
+                shape: partial_u8_bits_to_u8(&raw_face_style[1..=4]),
+                skin_color: partial_u8_bits_to_u8(&raw_face_style[5..=7]),
+            },
+            wrinkles: partial_u8_bits_to_u8(&raw_face_details[0..=3]),
+            makeup: partial_u8_bits_to_u8(&raw_face_details[4..=7]),
+        };
+
+        let hair_details = HairDetails {
+            style: raw_mii_data[0x32],
+            color: partial_u8_bits_to_u8(&raw_hair_details[0..=2]),
+            is_flipped: raw_hair_details[3],
+        };
+
+        let eye_details = EyeDetails {
+            style: partial_u8_bits_to_u8(&raw_eye_details[0..=5]),
+            color: partial_u8_bits_to_u8(&raw_eye_details[6..=8]),
+            scale: partial_u8_bits_to_u8(&raw_eye_details[9..=12]),
+            y_scale: partial_u8_bits_to_u8(&raw_eye_details[13..=15]),
+            rotation: partial_u8_bits_to_u8(&raw_eye_details[16..=20]),
+            x_spacing: partial_u8_bits_to_u8(&raw_eye_details[21..=24]),
+            y_position: partial_u8_bits_to_u8(&raw_eye_details[25..=29]),
+        };
+
+        let eyebrow_details = EyebrowDetails {
+            style: partial_u8_bits_to_u8(&raw_eyebrow_details[0..=4]),
+            color: partial_u8_bits_to_u8(&raw_eyebrow_details[5..=7]),
+            scale: partial_u8_bits_to_u8(&raw_eyebrow_details[8..=11]),
+            y_scale: partial_u8_bits_to_u8(&raw_eyebrow_details[12..=14]),
+            rotation: partial_u8_bits_to_u8(&raw_eyebrow_details[16..=19]),
+            x_spacing: partial_u8_bits_to_u8(&raw_eyebrow_details[21..=24]),
```

Comment:
Are there supposed to be gaps here?

Reply:
Yes, here it is

Reply:
![IMG_20221102_072428.jpg](https://user-images.githubusercontent.com/48295028/199413744-13a8a30d-80d2-42f6-a20f-518bd5eb1750.jpg)